src/main/java/com/timgroup/statsd/Message.java

```diff
@@ -138,7 +138,7 @@ public boolean equals(Object object) {
         if (object instanceof Message) {
             final Message msg = (Message)object;
 
-            boolean equals = (this.getAspect() == msg.getAspect())
+            boolean equals = (Objects.equals(this.getAspect(), msg.getAspect()))
```

Comment:
Good catch. Could you please also make a test for this? Maybe modify [aggregate_messages](https://github.com/DataDog/java-dogstatsd-client/blob/master/src/test/java/com/timgroup/statsd/StatsDAggregatorTest.java#L184) to pass some aspects as new objects.

Also, I guess we can call `.equals()` directly here:
```suggestion
            boolean equals = this.getAspect().equals(msg.getAspect())
```

Reply:
resolved, added one new message. Hope this was what you meant